public/app/features/dashboard-scene/panel-edit/VizPanelManager.tsx

```diff
@@ -128,6 +129,27 @@ export class VizPanelManager extends SceneObjectBase<VizPanelManagerState> {
       ...restOfOldState,
     });
 
+    // When changing from non-data to data panel, we need to add a new data provider
+    if (!restOfOldState.$data && !config.panels[pluginType].skipDataQuery) {
+      let ds = getLastUsedDatasourceFromStorage(getDashboardSceneFor(this).state.uid!)?.datasourceUid;
+
+      if (!ds) {
+        ds = config.defaultDatasource;
+      }
```

Comment:
@ivanortegaalba  is this logic correct? Should we first try to pick last used data source and only den fall back to the default one?